File: atcoder/ABC/097/B-Exponential.py
x = int(input())
ans = 1
diff = 1005
b = 2
while b * b <= x:
    bp = b
    while bp * b <= x:
        bp = bp * b
    if x - bp < diff:
        diff = x - bp
        ans = bp
    b += 1
print(ans)

File: atcoder/ABC/097/B-Exponential.cpp
#include <iostream>
using namespace std;

int main() {
    int X;
    cin >> X;

    int ans = 1, diff = 1005;
    for (int b = 2; b * b <= X; b++) {
        int bp = b;
        while (bp * b  <= X) 
            bp *= b;
        if (X - bp < diff) {
            diff = X - bp;
            ans = bp;
        }
    }
    cout << ans << endl;

    return 0;
}
